ymer/src/main/java/com/avanza/ymer/PersistedInstanceIdCalculationService.java

```diff
@@ -172,7 +196,10 @@ private void calculatePersistedInstanceId(String collectionName, Set<Integer> nu
 			log.info("Step 2/3\tWill delete no longer used fields [{}]", String.join(", ", noLongerNeededFields));
 		}
 
-		try (Stream<List<Document>> batches = StreamUtils.buffer(collection.findByQuery(createQuery(BATCH_SIZE, fieldNamesToCalculate, noLongerNeededFields)), BATCH_SIZE)) {
+		Query query = createQuery(BATCH_SIZE, fieldNamesToCalculate, noLongerNeededFields);
+		statistics.setTotalToCalculate(collection.countByQuery(query));
```

Comment:
Is it possible that the collection is modified in between the execution of this query and the next, which would lead to `totalToCalculate` being slightly wrong? What consequence would that lead to?

Reply:
Yes, of course you're right there 🤦‍♂️  Simply using a `boolean` to determine if the calculation is ready as you suggested might be much simpler.

Reply:
As this is now changed to a `boolean`, this problem is avoided.

Reply:
👍